### ISSUE TITLE
hono/jsx SSR performance

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#3347]

**Comment by yusukebe** [comment]
Hi @usualoma 

I've tried to do a benchmark of the in [KitaJS](https://github.com/kitajs/html) repository.

https://github.com/kitajs/html/tree/master/benchmarks

I've added hono/jsx in my forked repo: https://github.com/yusukebe/kitajs-html/tree/add-hono-jsx

If you are interested, you can run it and find that hono/jsx is slow in the "Real World" case. I never say we need to make it faster, but it's not good that hono/jsx is slower than React. However, I can't think of anything I can do right, you'll just have to take it as a reference. Thank you.

**Comment by usualoma** [comment]
Hi @yusukebe 
Thank you!
I think it is very slow since v4.5.0 (with React 19 Compatibility). I'll see if I can work around it.

I had the following results in my environment.

##### "^4.5.10"

```
benchmark        time (avg)             (min … max)       p75       p99      p999
--------------------------------------------------- -----------------------------
• Real World Scenario
--------------------------------------------------- -----------------------------
KitaJS/Html     435 µs/iter     (366 µs … 2'217 µs)    399 µs    838 µs  1'762 µs
Typed Html    1'641 µs/iter   (1'471 µs … 2'052 µs)  1'746 µs  2'024 µs  2'052 µs
VHtml         2'154 µs/iter   (2'018 µs … 2'669 µs)  2'229 µs  2'562 µs  2'669 µs
React JSX     5'680 µs/iter   (5'272 µs … 6'483 µs)  5'903 µs  6'454 µs  6'483 µs
Preact          801 µs/iter     (625 µs … 6'559 µs)    705 µs  1'764 µs  6'559 µs
React         5'594 µs/iter   (5'255 µs … 6'446 µs)  5'772 µs  6'392 µs  6'446 µs
Common Tags   2'687 µs/iter   (2'543 µs … 3'148 µs)  2'723 µs  3'122 µs  3'148 µs
Ghtml           195 µs/iter     (165 µs … 1'137 µs)    186 µs    730 µs  1'067 µs
JSXTE         3'749 µs/iter   (3'528 µs … 6'696 µs)  3'802 µs  4'105 µs  6'696 µs
hono/jsx      5'925 µs/iter   (5'347 µs … 6'382 µs)  6'017 µs  6'382 µs  6'382 µs

summary for Real World Scenario
  Ghtml
   2.23x faster than KitaJS/Html
   4.11x faster than Preact
   8.42x faster than Typed Html
   11.05x faster than VHtml
   13.78x faster than Common Tags
   19.24x faster than JSXTE
   28.7x faster than React
   29.14x faster than React JSX
   30.4x faster than hono/jsx
```

##### "4.4.13"

```
benchmark        time (avg)             (min … max)       p75       p99      p999
--------------------------------------------------- -----------------------------
• Real World Scenario
--------------------------------------------------- -----------------------------
KitaJS/Html     420 µs/iter     (344 µs … 1'406 µs)    380 µs    824 µs    967 µs
Typed Html    1'639 µs/iter   (1'459 µs … 2'120 µs)  1'744 µs  2'008 µs  2'120 µs
VHtml         2'149 µs/iter   (2'007 µs … 2'624 µs)  2'240 µs  2'554 µs  2'624 µs
React JSX     5'708 µs/iter   (5'346 µs … 6'810 µs)  5'927 µs  6'575 µs  6'810 µs
Preact          810 µs/iter     (627 µs … 2'151 µs)    722 µs  1'941 µs  2'151 µs
React         5'724 µs/iter   (5'363 µs … 6'573 µs)  5'921 µs  6'517 µs  6'573 µs
Common Tags   2'773 µs/iter   (2'558 µs … 3'611 µs)  2'821 µs  3'398 µs  3'611 µs
Ghtml           206 µs/iter     (167 µs … 1'288 µs)    193 µs  1'023 µs  1'239 µs
JSXTE         3'706 µs/iter   (3'259 µs … 4'934 µs)  3'787 µs  4'346 µs  4'934 µs
hono/jsx      1'616 µs/iter   (1'412 µs … 2'395 µs)  1'712 µs  2'374 µs  2'395 µs

summary for Real World Scenario
  Ghtml
   2.04x faster than KitaJS/Html
   3.94x faster than Preact
   7.85x faster than hono/jsx
   7.96x faster than Typed Html
   10.44x faster than VHtml
   13.47x faster than Common Tags
   18x faster than JSXTE
   27.72x faster than React JSX
   27.8x faster than React
```

**Comment by yusukebe** [comment]
Wooow, great!

**Comment by yusukebe** [comment]
Closed by #3352
